### DISCUSSION TITLE
continue to train the trainingsfile

### DISCUSSION THREAD

**Discussion by Flymaster01** [talmolab/sleap#1920] [Help!] [answered]
I want to train a lot videos with single interference mode. So i trained a file and got my predictions. But now the predictions are not completely correct. So I wanna continue to train my Trainingsfile with new videos. But how do I do that?

**Comment by roomrys** [comment, marked as answer]
Hi @Flymaster01,

There are a few routes you can go to get your model working on new videos:

---

### Option 1. (not advised) Create a new model by (initially) correcting predictions from the trained model
This option only uses the labeled frames from the new videos to train the model; thus the goal here is not to generalize the model, but to train another model specialized to these new videos. This is not generally the advised approach unless the videos are so different that it will be very difficult to create a generalized model.
1. Create a new project
2. Add the new video(s)
3. Generate ~20 Labeling Suggestions **on the new videos**
4. Predict on the Labeling Suggestions using the trained model
5. Correct the predictions
6. Generate ~20 Labeling Suggestions to "Predict On" after training
7. Train the model, opting to  "Predict On: suggested frames"
8. Repeat from step 5 until the pose estimation is sufficient

---

### Option 2. (recommended) Generalize the model by adding the new videos to the existing project
This option will use labeled frames from the existing project and from the new videos to essentially create a generalized model that predicts sufficiently across multiple videos.
1. Open the existing project
2. Add the new video(s)
3. Generate ~20 Labeling Suggestions **on the new videos**
4. Predict on the Labeling Suggestions using the trained model
5. Correct the predictions
6. Generate ~20 Labeling Suggestions to "Predict On" after training
7. Train the model, opting to  "Predict On: suggested frames"
8. Repeat from step 5 until the pose estimation is sufficient

---

### Option 3. Keep projects modular, but have one "omega" project that combines all projects
A third option that is really just a combination of the two above is to start with **Option 1** above so that you now have two projects with labels on "different enough" data. Then, create a third project which acts as your omega project that merges the two projects. This third project can then be used to train a generalized model and you still have each individual project giving you the option of training specialized models if wanted/needed. 
> The downfall of this option over over Option 2 is that it is not guaranteed that the model trained by merging the initial labels will actually be generalized as we had no feedback while correcting predictions; hence steps 5-7 below.

1. Complete the steps in Option 1
2. Open the original project
3. Select Save As and save in a new directory (for organization purposes)
4. Merge the "Option 1" project into the current project you just saved as (File > Merge Into Project...)
5. Generate ~20 Labeling Suggestions to "Predict On" after training
6. Train the model, opting to  "Predict On: suggested frames"
7. Repeat from step 5 until the pose estimation is sufficient

---

Let us know if you have any trouble.

Thanks,
Liezl

Related:
- #1311
